docs/book/user-guide/advanced-guide/client.md

```diff
@@ -0,0 +1,180 @@
+)
+
+for pipeline_run in my_runs_on_current_stack:
+    print(pipeline_run.name)
+```
+
+## List of Resources
+
+These are the main ZenML resources that you can interact with via the ZenML
+Client:
+
+### Pipelines, Runs, Artifacts
+- **Pipelines**: The [pipeline versions](../starter-guide/version-pipelines)
+that were implicitly tracked when running ZenML pipelines.
+- **Pipeline Runs**: Information about all pipeline runs that were executed on
+your ZenML instance.
+- **Step Runs**: The steps of all pipeline runs. Mainly useful for directly
+fetching a specific step of a run by its ID.
+- **Artifacts**: Information about all artifacts that were written to your
+artifact stores as part of pipeline runs.
+- **Schedules**: Metadata about the schedules that you have used to 
+[schedule pipeline runs](schedule-pipeline-runs).
+- **Builds**: The pipeline-specific Docker images that were created when
+[containerizing your pipeline](containerize-your-pipeline.md).
+- **Code Repositories**: The git code repositories that you have connected with
+your ZenML instance. See [here](connect-your-git-repository) for more information.
+
+{% hint style="info" %}
+Checkout the [documentation on fetching runs](../starter-guide/fetch-runs-after-execution.md)
+for more information on the various ways how you can fetch and use the
+pipeline, pipeline run, step run, and artifact resources in code.
+{% endhint %}
+
+### Stacks, Infrastructure, Authentication
+- **Stack**: The stacks registered in your ZenML instance.
+- **Stack Components**: The stack components registered in your ZenML instance,
+e.g., all orchestrators, artifact stores, model deployers, ...
+- **Flavors**: The [stack component flavors](../../getting-started/core-concepts.md#flavor)
+available to you, including:
+    - Built-in flavors like the [local orchestrator](../component-guide/orchestrators/local.md),
+    - Integration-enabled flavors like the [Kubeflow orchestrator](../component-guide/orchestrators/kubeflow.md),
+    - Custom flavors that you have [created yourself](../../platform-guide/set-up-your-mlops-platform/implement-a-custom-integration#step-2-create-individual-stack-component-flavors). 
+- **User**: The users registered in your ZenML instance. If you are running
+locally, there will only be a single `default` user. For more information on
+user management see [here](../../platform-guide/set-up-your-mlops-platform/user-management.md).
+- **Secrets**: The infrastructure authentication secrets that you have 
+registered in the 
+[ZenML Secret Store](../../platform-guide/set-up-your-mlops-platform/use-the-secret-store/use-the-secret-store.md).
+- **Service Connectors**: The service connectors that that you have set up to
+[connect ZenML to your infrastructure](../../platform-guide/set-up-your-mlops-platform/connect-zenml-to-infrastructure.md).
+
+## Client Methods
+
+### CRUD (Create, Read, Update, Delete)
+
+For most resources, the ZenML Client provides five different CRUD methods:
```

Comment:
Maybe don't talk about CRUD, since we interchangeably use create vs register, and as you mention there is an extra `list` option.. Feels like adds to confusion.